code/datums/components/defibrillator.dm

```diff
@@ -0,0 +1,382 @@
+/**
+ * A component for an item that attempts to defibrillate a mob when activated.
+ */
+/datum/component/defib
+	/// If this is being used by a borg or not, with necessary safeties applied if so.
+	var/robotic
+	/// If it should penetrate space suits
+	var/combat
+	/// If combat is true, this determines whether or not it should always cause a heart attack.
+	var/combat_defib_chance
+	/// Whether the safeties are enabled or not
+	var/safety
+	/// If the defib is actively performing a defib cycle
+	var/busy = FALSE
+	/// Cooldown length for this defib in seconds
+	var/cooldown
```

Comment:
```suggestion
	/// Cooldown length for this defib in deciseconds
	var/cooldown
```
the way you use this var suggests that you treat this var like its unit is deciseconds, obv values should be assigned with the `SECONDS` define but that's still in the end measured in deciseconds